scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -572,28 +571,26 @@ private void satisfyGoalGeneral(Goal goal) {
     //prune based on constraints on goal and activity type (mutex, state,
     //event, etc)
     //TODO: move this into polymorphic method. don't want to be demuxing types
-    Collection<StateConstraintExpression> stateConstraints = new LinkedList<>();
+    Collection<Expression<Windows>> resourceConstraints = new LinkedList<>();
 
     //add all goal constraints
-    StateConstraintExpression goalConstraints = goal.getStateConstraints();
-    ActivityType actType;
+    final var goalConstraints = goal.getResourceConstraints();
 
     if (goalConstraints != null) {
-      stateConstraints.add(goalConstraints);
+      resourceConstraints.add(goalConstraints);
     }
     if (missing instanceof final MissingActivityInstanceConflict missingInstance) {
       final var act = missingInstance.getInstance();
-      actType = act.getType();
-      StateConstraintExpression c = act.getType().getStateConstraints();
-      if (c != null) stateConstraints.add(c);
+      final var c = act.getType().getStateConstraints();
+      if (c != null) resourceConstraints.add(c);
     } else if (goal instanceof ActivityTemplateGoal) {
```

Comment:
Not a blocker - just a style suggestion - instead of casting to `((ActivityTemplateGoal) goal)` below, we can define a new `g` variable like so:

```suggestion
    } else if (goal instanceof ActivityTemplateGoal g) {
```